experiment/measurer/measure_worker.py

```diff
@@ -78,6 +80,49 @@ def extract_corpus(corpus_archive: str, sha_blacklist: Set[str],
         filesystem.write(file_path, member_contents, 'wb')
 
 
+class StateFile:
+    """A class representing the state of measuring a particular trial on
+    particular cycle. Objects of this class are backed by files stored in the
+    bucket."""
+
+    def __init__(self, name: str, state_dir: str, cycle: int):
+        self.name = name
+        self.state_dir = state_dir
+        self.cycle = cycle
+        self._prev_state = None
+
+    def _get_gcs_cycle_state_file_path(self, cycle: int) -> str:
+        state_file_name = experiment_utils.get_cycle_file_name(
+            self.name, cycle, '.json')
+        state_file_path = os.path.join(self.state_dir, state_file_name)
+        return exp_path.gcs(pathlib.Path(state_file_path))
+
+    def _get_previous_cycle_state(self, first_cycle_default) -> str:
```

Comment:
Optional nit: It also seems unlikely that we'd ever want anything besides an empty list as the value for our default in the first cycle case. Should that also be omitted?

Reply:
+1. this first_cycle_Default param is confusing.

Reply:
I now return an empty list on first cycle. I think the first_cycle_default thing would be necessary if these methods ever returned anything besides a list, but since they don't in practice the empty list is fine.